atomic_reactor/plugins/pre_pull_base_image.py

```diff
@@ -62,16 +63,21 @@ def run(self):
 
             base_image_with_registry.registry = self.parent_registry
 
-        pulled_base = self.tasker.pull_image(base_image_with_registry,
-                                             insecure=self.parent_registry_insecure)
-        if (base_image_with_registry.namespace != 'library' and
-                not self.tasker.image_exists(base_image_with_registry.to_str())):
-            self.log.info("'%s' not found", base_image_with_registry.to_str())
-            base_image_with_registry.namespace = 'library'
-            self.log.info("trying '%s'", base_image_with_registry.to_str())
-            pulled_base = self.tasker.pull_image(base_image_with_registry,
-                                                 insecure=self.parent_registry_insecure)
+        try:
+            self.tasker.pull_image(base_image_with_registry,
+                                   insecure=self.parent_registry_insecure)
 
+        except RetryGeneratorException:
+            if base_image_with_registry.namespace != 'library':
+                self.log.info("'%s' not found", base_image_with_registry.to_str())
+                base_image_with_registry.namespace = 'library'
+                self.log.info("trying '%s'", base_image_with_registry.to_str())
+                self.tasker.pull_image(base_image_with_registry,
+                                       insecure=self.parent_registry_insecure)
+            else:
+                raise
```

Comment:
Could put the `raise` block first to avoid needing to indent the rest, e.g.
```
if ... == 'library':
    # Nothing else to try
    raise

self.log.info(...)
...
```

Also, I think we want the error to look different for these two cases:
- retries failed due to some sort of network issue (this is a potentially serious system issue)
- the image just isn't there (this is just user error)

At the very least the exception's message should be different... even better would be a different exception. We seem to use RuntimeError for that sort of thing, but even that seems pretty generic.

Reply:
ok will put raise block first